### PR TITLE
Added convergence criteria option to input XML

### PR DESCRIPTION
**Pull request by j-bryan**
--------
Pull Request Description
--------
##### What issue does this change request address?
#162

##### What are the significant changes in functionality due to this change request?
Added the ability to add a node for optimization settings in the HERON input XML file which modifies the default values in the template when generating outer.xml.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/HERON/wiki/Code-Standards) for details.
- [x] 4. Automated Tests should pass.
- [x] 5. If significant functionality is added, there must be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large tes.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added, the the analytic documentation must be updated/added.
- [x] 9. If any test used as a basis for documentation examples have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

File: templates/template_driver.py [hunk 1]
```diff
@@ -342,7 +342,7 @@ def _modify_outer_dataobjects(self, template, case, components):
     elif case.get_mode() == 'opt':
       self._remove_by_name(DOs, ['grid'])
     # update optimization settings if provided
-    if (case.get_mode() == 'opt') and (case._optimization_settings is not None):
+    if (case.get_mode() == 'opt') and (case._optimization_settings is not None) and (not case.debug['enabled']):
```
Comment (dylanjm) [inline templates/template_driver.py]: I wonder if we should leave these out for another PR. I feel like there should be a different way to prevent that error from happening. @PaulTalbot-INL What are your thoughts? This is in regards to #158.

Comment (PaulTalbot-INL) [reply templates/template_driver.py]: I think this is a fine fix now; perhaps adding a comment and not close the issue would let us find a better way to handle it overall.

I will admit I'm not super happy with the order of operations dependence in the template driver; I don't know a better way offhand to implement it though.

**Review by dylanjm — APPROVED** [review]
